packages/kbn-capture-oas-snapshot-cli/src/capture_oas_snapshot.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fs from 'node:fs/promises';
+import { encode } from 'node:querystring';
+import type { ChildProcess } from 'node:child_process';
+import fetch from 'node-fetch';
+import * as Rx from 'rxjs';
+import { startTSWorker } from '@kbn/dev-utils';
+import { createTestEsCluster } from '@kbn/test';
+import type { ToolingLog } from '@kbn/tooling-log';
+import { createTestServerlessInstances } from '@kbn/core-test-helpers-kbn-server';
+import type { Result } from './kibana_worker';
+import { sortAndPrettyPrint } from './run_capture_oas_snapshot_cli';
+import { buildFlavourEnvArgName } from './common';
+
+interface CaptureOasSnapshotArgs {
+  log: ToolingLog;
+  buildFlavour: 'serverless' | 'traditional';
+  outputFile: string;
+  update: boolean;
+  filters?: {
+    pathStartsWith?: string[];
+    excludePathsMatching?: string[];
+  };
+}
+
+const MB = 1024 * 1024;
+const twoDeci = (num: number) => Math.round(num * 100) / 100;
+
+export async function captureOasSnapshot({
+  log,
+  filters = {},
+  buildFlavour,
+  update,
+  outputFile,
+}: CaptureOasSnapshotArgs): Promise<void> {
+  const { excludePathsMatching = [], pathStartsWith } = filters;
+  // internal consts
+  const port = 5622;
+  // We are only including /api/status for now
+  excludePathsMatching.push(
+    '/{path*}',
+    // Our internal asset paths
+    '/XXXXXXXXXXXX/'
```

Comment:
Are we sure we will always only use the tool in dev mode? because the dev asset path won't work with any Kibana build, correct?